### PR TITLE
Chapter 3 - ISA: X86 duality 

### PR DESCRIPTION
**Pull request by pveentjer**
Added note about the duality of load/store and register/memory behavior of the X86.

### CODE REVIEW

**Comment by dendibakh** [comment]
Not sure what value it provides to the readers. If there is some implicit point that you're trying to make, then I suggest to make it explicit.

**Comment by pveentjer** [comment]
Hi Denis,

thank you for the review.

The point I'm trying to make is that in the initial section, it states that modern architectures are load-store. X86 is one of the most used ISAs and isn't a load-store architecture but register-memory. As a consequence, a reader of the book could falsely assume, that X86 is a load-store architecture.

If you think this brings no value to the book, I'll close the PR. 

If you think it could bring value, I'm all ears to rewrite it. Perhaps add it as a footnote?

Regards,

Peter.

**Comment by pveentjer** [comment]
Hi Denis,

the original text was correct. So modern ISA's like RISC-V and ARM are load-store architectures. 

The X86 ISA is register memory, but after uops conversion, the X86 microarchitecture also has transformed into a load-store architecture.

**Comment by dendibakh** [comment]
Ooops, of course, you're right. I was implicitly thinking about x86 again. :)
Will fix.

**Comment by dendibakh** [comment]
Please check now.

**Comment by pveentjer** [comment]
What is missing is that the X86 microarchitecture is a load/store architecture as part of the uops conversion.

Something like this:
```
inc [A],[B],[C]
```
will be converted into

```
load r1, [A]
load r2, [B]
inc r3, r1, r2
store r3, [C]
```

I find it very helpful because I need to think a lot less about the complex addressing modes and it helps me to understand the opportunities for performance. In the first example, it isn't immediately clear that the loads of [A] and [B] can be parallelized, but in the uops version, it is much easier to understand.

**Comment by dendibakh** [comment]
Ok, I agree, but I think this section is not the best place to discuss uops. I have a section for this: `4-4 UOP.md`
There I talk about uops cracking. Please check it and let me know if you have any comments.

File: chapters/4-Terminology-And-Metrics/4-4 UOP.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ typora-root-url: ..\..\img
 
 ## Micro-ops {#sec:sec_UOP}
 
-Microprocessors with the x86 architecture translate complex CISC-like instructions into simple RISC-like microoperations, abbreviated as $\mu$ops or $\mu$ops. A simple addition instruction such as `ADD rax, rbx` generates only one $\mu$op, while a more complex instruction like `ADD rax, [mem]` may generate two: one for reading from the `mem` memory location into a temporary (un-named) register, and one for adding it to the `rax` register. The instruction `ADD [mem], rax` generates three $\mu$ops: one for reading from memory, one for adding, and one for writing the result back to memory.
```
Comment (pveentjer) [inline chapters/4-Terminology-And-Metrics/4-4 UOP.md:7]: I have added the comment about load-store architecture to this section. I have also renamed 'reading' and 'writing' to 'load' and 'store' to use the same terms as load-store architecture.

wdyt?

**Comment by dendibakh** [comment]
Thank you @pveentjer !

**Pull request merged**